marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/impl/process.xqy

```diff
@@ -251,19 +252,22 @@ declare function proc-impl:process-match-and-merge-with-options(
           map:entry("context",
             map:new((
               map:entry("collections",
-                coll-impl:on-merge(
-                  map:new((
-                    for $uri in $distinct-uris
-                    let $write-object := proc-impl:retrieve-write-object($write-objects-by-uri, $uri)
-                    return
-                      map:entry(
-                        $uri,
-                        $write-object
-                          => map:get("context")
-                          => map:get("collections")
-                      )
-                  )),
-                  $on-merge-options
+                (
```

Comment:
I looked at the function that this code is in - it's about 250 lines, and of course a very important one. I think when we change it, it'd be nice to extract smaller functions to make it easier to understand, and then we can write unit tests for those smaller functions. Perhaps this could be a "build-collections-for-merged-uri" function? Or "build-context-for-merged-uri", which would include the permissions too? Or "build-merge-result" for the entire map? (I like the verb "build" for anything that builds something in-memory but doesn't have save it (I like "create" when something is saved)).

Reply:
Can we handle that in a technical task ticket separate from this bug fix?